libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/PredictiveCache.kt

```diff
@@ -48,28 +51,49 @@ object PredictiveCache {
     }
 
     fun createMapsController(
+        mapboxMap: Any,
         tileStore: TileStore,
         tileVariant: String,
-        predictiveCacheLocationOptions: PredictiveCacheLocationOptions =
-            PredictiveCacheLocationOptions.Builder().build()
+        predictiveCacheLocationOptions: PredictiveCacheLocationOptions
     ) {
         val predictiveCacheController =
             MapboxNativeNavigatorImpl.createMapsPredictiveCacheController(
                 tileStore,
                 tileVariant,
                 predictiveCacheLocationOptions
             )
-        cachedMapsPredictiveCacheControllers[tileVariant] = predictiveCacheController
-        mapsPredictiveCacheLocationOptions[tileVariant] =
-            Pair(tileStore, predictiveCacheLocationOptions)
+
+        val cacheControllers = cachedMapsPredictiveCacheControllers[mapboxMap] ?: mutableMapOf()
+        cacheControllers[tileVariant] = predictiveCacheController
+        cachedMapsPredictiveCacheControllers[mapboxMap] = cacheControllers
+
+        val locationOptions = mapsPredictiveCacheLocationOptions[mapboxMap] ?: mutableMapOf()
+        locationOptions[tileVariant] = Pair(tileStore, predictiveCacheLocationOptions)
+        mapsPredictiveCacheLocationOptions[mapboxMap] = locationOptions
```

Comment:
Why do we need to keep a reference to location options? Isn't it enough to pass them when creating the controller?

Reply:
if navigator is recreated, we need to recreate all controllers, so we need all options